pandas/tests/frame/methods/test_replace.py

```diff
@@ -1503,3 +1503,10 @@ def test_regex_replace_scalar(
 
         expected.loc[expected["a"] == ".", "a"] = expected_replace_val
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("regex", [False, True])
+    def test_replace_regex_dtype_frame(self, regex):
+        df = DataFrame({"A": ["0"], "B": ["0"]})
```

Comment:
in this case both columns will get cast.  can you do have another case where only one column is affected

Reply:
Done.

Added extra testcase.